doc/design/dist/README.md

```diff
@@ -0,0 +1,78 @@
+# Distributed Training Design Doc
+
+## Objective
+
+We want Paddle to support training on a general-purpose cluster. The cluster runs Paddle, the Web server (e.g., Nginx), the log collector (e.g., fluentd), the distributed queue service (e.g., Kafka), the log joiner and other data processors written using Storm, Spark, and Hadoop MapReduce on the same cluster. As illustrated in the following graph:
+
+![general purpose cluster](src/arch.png)
+
+This poses new challenges for Paddle,
+
+- Paddle need to be tault tolerant.
+- Input training data can be online data from realtime logs, or batched data from distributed file system.
+- User needs a simple way to train model on cloud. Complexities such as job scheduling should be hidden from user.
+
+## Training Job
+
+A training job will be created once user asks Paddle cloud to train a model. The training job is made up of different processes that collabratively consume data input and produce a trained model. There are three kind of processes:
+
+- Master process
+- Trainer process
+- Parameter server process
+
+One training job will only have one master process, typicall multiple trainer and parameter server processes. Their relation is illustrated in the following graph:
+
+![process collabration](src/paddle-on-kubernetes-invited-blog-model-sharding.png)
+
+### Master Process
+
+Master process will:
+
+- keep a list of alive trainers and a list of alive parameter servers and do *health check*,
+  - if trainer is dead it will update task queue accordingly as mentioned in [task queue](#task-queue).
+  - if a parameter server is dead or a new parameter server joins, it will broacast this information to all trainers.
```

Comment:
It seems that we shouldn't allow a changeable number of parameter servers, because every time this number changes, all trainer processes need to re-split their local models so to correspond model shards to the parameter servers.